src/test/editor-service/editor-server-test.ts

```diff
@@ -0,0 +1,155 @@
+      await getNextResponse(0);
+      await sendRequest({id: 1, value: initMessage});
+      const errorMessage = await invertPromise(getNextResponse(1));
+      assert.equal(errorMessage, 'Already initialized!');
+    });
+    test('the first request must be initialization', async function() {
+      await sendRequest({id: 0, value: getWarningsMessage});
+      const errorMessage = await invertPromise(getNextResponse(0));
+      assert.equal(
+          errorMessage,
+          `Must send an 'init' message before any others. Received ` +
+              `'getWarningsFor' message before 'init'.`);
+    });
+    const testName = 'can perform editor service functions once initialized';
+    test(testName, async function() {
+      await sendRequest({id: 0, value: initMessage});
+      await getNextResponse(0);
+      await sendRequest({id: 1, value: getWarningsMessage});
+      const warnings = await getNextResponse(1);
+      assert.deepEqual(warnings, [{
+                         code: 'parse-error',
+                         message: 'Unexpected token <',
+                         severity: 0,
+                         sourceRange: {
+                           file: 'malformed.html',
+                           start: {line: 266, column: 0},
+                           end: {line: 266, column: 0}
+                         }
+                       }]);
+    });
+  }
+
+  let suiteName =
+      'from node with child_process.fork() and process.send() for IPC';
+  suite(suiteName, function() {
+    let child: child_process.ChildProcess;
+    setup(() => {
+      child = child_process.fork(pathToServer);
+    });
+    teardown(() => {
+      child.kill();
+    });
+
+    async function sendRequest(request: any) {
+      child.send(request);
+    };
+
+    async function getNextResponse(expectedId: number) {
+      const message = await new Promise<any>((resolve) => {
+        child.once('message', function(msg: any) {
+          resolve(msg);
+        });
+      });
+      assert.equal(message.id, expectedId);
+      return new Promise((resolve, reject) => {
+        if (message.value.kind === 'resolution') {
+          resolve(message.value.resolution);
+        } else if (message.value.kind === 'rejection') {
+          reject(message.value.rejection);
+        }
+      });
+    };
+
+    editorServiceInterfaceTests(sendRequest, getNextResponse);
+  });
+
+  suiteName = 'from the command line with stdin and stdout';
+  suite(suiteName, function() {
+    let child: child_process.ChildProcess;
+    let lines: NodeJS.ReadableStream;
+    setup(() => {
+      child = child_process.spawn(
+          'node', [pathToServer], {stdio: ['pipe', 'pipe', 'pipe']});
+      child.stdout.setEncoding('utf8');
+      child.stdout.resume();
+      lines = child.stdout.pipe(split());
+    });
+    teardown(() => {
+      child.kill();
+    });
+
+    async function sendRequest(message: any) {
+      return new Promise<void>((resolve, reject) => {
```

Comment:
Since you're in an async function, throw an error rather than create a new Promise.